lib/scala/kg/src/main/scala/formats/kg/kgtk/KgtkTsvReader.scala

```diff
@@ -0,0 +1,71 @@
+package formats.kg.kgtk
+
+import formats.CsvReader
+import com.github.tototoshi.csv.{CSVReader, TSVFormat}
+import models.kg.KgEdge
+import models.kg.KgNode
+import org.slf4j.LoggerFactory
+import java.nio.file.Path
+import java.io.InputStream
+
+final class KgtkTsvReader(csvReader: CSVReader) extends CsvReader[Tuple3[KgEdge, KgNode, KgNode]](csvReader) {
+  private final val LIST_DELIM = "|";
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  def iterator: Iterator[Tuple3[KgEdge, KgNode, KgNode]] =
+    csvReader.iteratorWithHeaders.map(row => {
+      val node1Labels = row.getList("node1;label", LIST_DELIM)
+      val node2Labels = row.getList("node2;label", LIST_DELIM)
+      Tuple3(
+        KgEdge(
+          datasource = row("datasource"),
+          datasources = row.getList("datasource", "|"),
+          id = row("id"),
+          `object` = row("node2"),
+          other = None,
+          predicate = row("relation"),
+          subject = row("node1"),
+          weight = row.getNonBlank("weight").flatMap(weight => {
```

Comment:
.flatMap is OK, but this block could use Try
https://docs.scala-lang.org/overviews/scala-book/functional-error-handling.html#trysuccessfailure
and its .toOption.
so something like
`Try { Integer.parseInt(s.trim) }.toOption`
This is a common pattern.

Reply:
Neat changed